lib/tmp.js

```diff
@@ -220,6 +220,14 @@ function _createTmpFile(options, callback) {
     fs.open(name, CREATE_FLAGS, opts.mode || FILE_MODE, function _fileCreated(err, fd) {
       if (err) return cb(err);
 
+      if (opts.discardDescriptor) {
+        return fs.close(fd, function _invokeCallback(err) {
+          cb(null, name, undefined, _prepareTmpFileRemoveCallback(name, -1, opts));
```

Comment:
do you know about sinonjs? you could temporarily replace fs.close with a mock that would then call the callback with a fake error.